src/d3/ForceGraph.tsx

```diff
@@ -1,22 +1,37 @@
 import * as d3 from "d3";
 import { GraphData, GraphNode, GraphLink } from "../Types/GraphTypes";
+import { HaltingFailure } from "../util/HaltingAnalysis";
 
 const SimValues = {
   ManyBodyStrength: -700
 };
 
 const NodeStyles = {
   radius: 6,
-  strokeWidth: 2,
-  fill: (n: SimNode, i: number) => (n.active ? "#0f0" : "#555"),
-  stroke: (d: SimNode, i: number) => {
-    return i === 0 ? "green" : "white";
+  className: (n: SimNode) => {
+    const classes = ["node"];
+    if (n.data.distance === 0) classes.push("self");
+    if (n.vulnerable) classes.push("vulnerable");
+    classes.push(n.live ? "live" : "dead");
+    return classes.join(" ");
   }
 };
 
 const LinkStyles = {
-  strokeWidth: (link: SimLink, i: number) => (link.active ? 2 : 0.2),
-  stroke: (link: SimLink, i: number) => (link.active ? "#0f0" : "#333"),
+  strokeWidth: (link: SimLink, i: number) => {
```

Comment:
Should we move the styling attributes to `ForceGraph.css` and use the logic here for basically defining the class names (similar to what you've done to `NodeStyles`)? 

I feel like having the style info split in two files might make it harder to find things in the future.